演習問題3－2～3－18

ddt-miyaoka opened the pull request.

ddt-miyaoka committed:
演習問題3－2～3－18

ddt-miyaoka committed:
metadata削除

ddt-mentor commented on ensyu3-11/src/Ensyu3_11/Ensyu3_11.java:
```diff
@@ -0,0 +1,45 @@
+		}else if (firstNumber == secondNumber){
+			//二つの数字の差は0となる
+			differenceNumber  = 0;
+		}
```
柴田
同じ値どうしで数値を減算すれば０になるはずで、
上記二つの分岐の内どちらかに混ぜ込めませんか？
※あえて3つに分けている理由があれば説明をお願いします。

ddt-mentor replied:
コメント記入者：卜部
少々、柴田くんの指摘と趣旨がズレているようです。
この場合、同値であると差分を算出することなく、あくまで差分保持用の変数に格納されている「初期値」が表示されるに過ぎません。
もし、初期値が0でなかったら成立しないようなロジックは避けるようにしてください。

ddt-mentor commented on ensyu3-16/src/Ensyu3_16/Ensyu3_16.java:
```diff
@@ -0,0 +1,79 @@
+		//真ん中の値を表示
+		System.out.println(mediumNumber);
+		//一番大きい値を表示
+		System.out.println(maxNumber);
```
柴田
入力値、1,2,3で試したところ、1,3,2と結果が返ってきました。
修正をお願いします。

ddt-mentor commented on ensyu3-17/src/Ensyu3_17/Ensyu3_17.java:
```diff
@@ -0,0 +1,30 @@
+		//ランダムで取得した値を用いてSwitch文を実行
+		switch(switchNumber){
+		//0の場合はグーを表示
+		case 0: System.out.println("グー");
```
柴田
読みづらくなるため、処理が単純でも、条件と処理は行を分けておくようにしてください。

ddt-mentor commented on ensyu3-2/src/Ensyu3_2/Ensyu3_2.java:
```diff
@@ -0,0 +1,30 @@
+		System.out.println("整数B：");
+		//入力した値を代入する
+		int secondNumber = stdIn.nextInt();
+		//もしfirstNumer割るsecondNumberが0であれば
```
柴田
微妙にコメントが異なるようですので確認してください。
たしかに％は除算するのですが、単に割るだけではありません。

ddt-mentor commented on ensyu3-3/src/Ensyu3_3/Ensyu3_3.java:
```diff
@@ -0,0 +1,29 @@
+package Ensyu3_3;
+//List3-5の最後のelseを、else if(n == 0)に変更したらどうなるかを検証せよ
```
柴田
検証した結果をコメントで良いので記載してください。
※こちらのレビューコメントに対する返信でも構いません

ddt-miyaoka replied:
0を入れた場合、elseの時と同じように値が0であると表示します

ddt-mentor commented on ensyu3-8/src/Ensyu3_8/Ensyu3_8.java:
```diff
@@ -0,0 +1,32 @@
+		}else if (checkNumber >= 80){
+			//条件に合致した場合優と表示する
+			System.out.println("優");
+		}
```
柴田
100を超えた場合はどうなりますか？
独自のもので構いませんので、自分で仕様を検討し、対応してください。
※仕様の内容は記載しておいてください。

ddt-miyaoka replied:
問題は成績の判定であり、最大値は100までです。
100を超えた場合は範囲外の値として扱うように処理を追記しました

ddt-miyaoka committed:
Update Ensyu3_8.java

入力した値が100を超えた場合の処理を追加。
成績の判定の場合は100が最大値であり、それを超えることはないため、100以上の値を入力した場合は範囲外の値として扱う

ddt-miyaoka committed:
PRの指摘内容の修正

ddt-miyaoka committed:
Merge branch 'E-03' of https://github.com/ddt-miyaoka/meikai_java_miyaoka into E-03

ddt-mentor commented on ensyu3-11/src/Ensyu3_11/Ensyu3_11.java:
```diff
@@ -26,10 +26,6 @@ public static void main(String[] args) {
-		}else if (firstNumber == secondNumber){
-			//二つの数字の差は0となる
-			differenceNumber  = 0;
 		}
```
コメント記入者：卜部
少々、柴田くんの指摘と趣旨がズレているようです。
この場合、同値であると差分を算出することなく、あくまで差分保持用の変数に格納されている「初期値」が表示されるに過ぎません。
もし、初期値が0でなかったら成立しないようなロジックは避けるようにしてください

ddt-miyaoka committed:
3-11修正
